include/usb/class/hid.h

```diff
@@ -0,0 +1,630 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ * Copyright (c) 2018,2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
```

Comment:
hid descriptors are not used in the usb only
many of the things provided here are also not very low level

The question is why this should stay in include/usb/class/hid.h? Maybe it should go to some more generic place include/input/hid_desc.h?

I think there is a trap in front of Zephyr. Unlike Linux it does not have clear separation between kernel and middleware/application layer.

Reply:
The location is something to discuss. My point is that the specification and all possible usages come from USB-IF and therefore I think `incluse/usb/class` fits better. I am also looking at HID Sensor Usages and I think placing this header in _include/input_ is probably not quite correct.

Reply:
Since the standard went beyond the usb and is used in other places (e.g. ble hid) it might be good to look for some common place now (since the code is being moved). But that goes only if there is such a place and it may be that there is none.

The good thig is that these definitions are separated so can be commonly used regardless of the location.

Reply:
I actually think keeping it in `include/usb/class` it a good fit, since most other uses refers to the USB standard as the origin of the HID protocol.